File: frontend-admin/lib/currency.ts
// Store currency, set per client at build time via NEXT_PUBLIC_CURRENCY_CODE (e.g. GBP, USD, INR).
const SYMBOLS: Record<string, string> = {
  GBP: "£", USD: "$", EUR: "€", INR: "₹",
  AUD: "A$", CAD: "C$", AED: "د.إ", SGD: "S$", NZD: "NZ$",
}

const CODE = (process.env.NEXT_PUBLIC_CURRENCY_CODE ?? "GBP").toUpperCase()

export const CURRENCY_SYMBOL = SYMBOLS[CODE] ?? `${CODE} `

/** Format an amount with the store currency symbol, e.g. "£12.00" / "$12.00". */
export function formatMoney(amount: number | string): string {
  const n = typeof amount === "string" ? parseFloat(amount) : amount
  return `${CURRENCY_SYMBOL}${(Number.isFinite(n) ? n : 0).toFixed(2)}`
}

File: backend/app/shared/currency.py
"""Store currency helpers. The currency is set per client via settings.CURRENCY_CODE."""
from app.core.config import settings

# Symbol per ISO 4217 code. Unknown codes fall back to "<CODE> " (e.g. "ZAR 12.00").
_SYMBOLS = {
    "GBP": "£", "USD": "$", "EUR": "€", "INR": "₹",
    "AUD": "A$", "CAD": "C$", "AED": "د.إ", "SGD": "S$", "NZD": "NZ$",
}


def currency_symbol() -> str:
    code = settings.CURRENCY_CODE.upper()
    return _SYMBOLS.get(code, f"{code} ")


def format_money(amount) -> str:
    """Format an amount with the store's currency symbol, e.g. '£12.00' / '$12.00'."""
    return f"{currency_symbol()}{float(amount):.2f}"
